src/ffi/extra/structural_equal.cc

```diff
@@ -255,7 +255,7 @@ class StructEqualHandler {
     }
   }
 
-  bool CompareMap(Map<Any, Any> lhs, Map<Any, Any> rhs) {
+  bool CompareMap(const Map<Any, Any>& lhs, const Map<Any, Any>& rhs) {
```

Comment:
under this paticular context, non-const ref is better, mainly because caller always move into the